### ISSUE TITLE
HSegmentControlDataSource Extension?

### ISSUE THREAD

**Issue by miller-ms** [popodidi/HSegmentControl#1]
First, thank you for making this control available.    Can you explain the purpose of the extension `HSegmentControlDataSource`?

I thought it's purpose was so an adopter of the `HSegmentControlDataSource` protocol would have a default implementation for the method: 

`func segmentControl(_ segmentControl: HSegmentControl, segmentBackgroundViewOfIndex index: Int) -> UIView.`

However, I always need to supply the method in the class that adopts the protocol.  How would a consumer use this extension?

**Comment by popodidi** [comment]
Thanks for this feedback. 
The extension is for the default implementation just as you said. It should be my mistake not to have the extension `public`. I just fixed it and updated to `2.1.2`. 
Please check if it works.

**Comment by miller-ms** [comment]
Yes, that works. thanks.  Will you be posting this update to cocoa pods?

**Comment by miller-ms** [comment]
Any chance you would be interested in updating this control to accept drag and swipe gestures to scroll through the list?

**Comment by popodidi** [comment]
It's already pushed to cocoa pods, `pod update` will do the trick.

Scrolling is possible, but it would be a rather huge amount of works under current implementation, which is not so nice actually.
The current implementation doesn't leverage any power of `UIScrollView` to show the animation. I simply wrap a long (in x-direction) `baseView: UIView` inside the `HSegmentControl` and set the x position to show the right segment.

Under this implementation, I would suggest two ways to have the scrolling functionality
#### UIScrollView
It seems to be more reasonable to wrap the `baseView` inside a `scrollView`, which fills the `HSegmentControl`, and leverage the power of `UIScrollView` to have the dragging/swiping functionality. In this case, the selection animation and x position `func`s should be rewritten or replaced.
#### UIGestureRecognizer
Simply add `UISwipeGestureRecognizer` to `HSegmentControl` and select the next/previous one on user swipes. It is easier to code, but not so intuitive in user interaction in my opinion.

So I would choose the first approach, but I am afraid I don't have enough time to update it in the near future. If you are interested, please feel free to update and pull request.
